static/app/views/pullRequest/details/main/pullRequestDetailsSizeContent.tsx

```diff
@@ -0,0 +1,120 @@
+import {useState} from 'react';
+import styled from '@emotion/styled';
+
+import {CompactSelect} from 'sentry/components/core/compactSelect';
+import {Flex, Grid, Stack} from 'sentry/components/core/layout';
+import {Heading} from 'sentry/components/core/text';
+import {t} from 'sentry/locale';
+import {useApiQuery, type UseApiQueryResult} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useParams} from 'sentry/utils/useParams';
+import {BuildDetailsMainContent} from 'sentry/views/preprod/buildDetails/main/buildDetailsMainContent';
+import {BuildDetailsSidebarContent} from 'sentry/views/preprod/buildDetails/sidebar/buildDetailsSidebarContent';
+import type {AppSizeApiResponse} from 'sentry/views/preprod/types/appSizeTypes';
+import type {BuildDetailsApiResponse} from 'sentry/views/preprod/types/buildDetailsTypes';
+
+interface PullRequestDetailsSizeContentProps {
+  buildDetails: BuildDetailsApiResponse[];
+}
+
+export function PullRequestDetailsSizeContent({
+  buildDetails,
+}: PullRequestDetailsSizeContentProps) {
+  const organization = useOrganization();
+  const {projectId} = useParams<{
+    projectId: string;
+  }>();
+  const [selectedBuildId, setSelectedBuildId] = useState<string | undefined>(
+    buildDetails[0]?.id
+  );
+
+  const appSizeQuery: UseApiQueryResult<AppSizeApiResponse, RequestError> =
+    useApiQuery<AppSizeApiResponse>(
+      [
+        `/projects/${organization.slug}/${projectId}/files/preprodartifacts/${selectedBuildId}/size-analysis/`,
+      ],
+      {
+        staleTime: 0,
+        enabled: !!projectId && !!selectedBuildId,
+      }
+    );
+
+  if (buildDetails.length === 0 || !selectedBuildId) {
+    return <div>No build details found</div>;
+  }
+
+  const getBuildDetailsLabel = (buildDetail: BuildDetailsApiResponse) => {
+    const {id, app_info} = buildDetail;
+    let label = `#${id}`;
+
+    if (app_info?.app_id) {
+      label += ` ${app_info.app_id}`;
+    }
+    if (app_info?.version) {
```

Comment:
I think this could run into some truthy issues, might want to explicitly check for null here.